### PR TITLE
Add Instruction Condition Implementation

### PR DESCRIPTION
**Pull request by kharland**

### CODE REVIEW

File: arm7_tdmi/lib/src/instruction/arm/condition.dart [hunk 1]
```diff
@@ -0,0 +1,253 @@
+import 'package:arm7_tdmi/src/cpu/program_status_registers.dart';
+import 'package:arm7_tdmi/src/utils/bits.dart';
+
+/// A bit string for determining whether an ARM instruction will execute.
+///
+/// A [Condition] is derived from bits 31-28 of an ARM instruction.
+abstract class Condition {
+  static final Condition equal = new _Equal();
```
Comment (matanlurey) [inline arm7_tdmi/lib/src/instruction/arm/condition.dart]: Make all of these const and then static const.

File: arm7_tdmi/lib/src/instruction/arm/condition.dart [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+  static final Condition unsignedHigher = new _UnsignedHigher();
+  static final Condition unsignedLowerOrSame = new _UnsignedLowerOrSame();
+  static final Condition greaterThanOrEqual = new _GreaterThanOrEqual();
+  static final Condition lessThan = new _LessThan();
+  static final Condition greaterThan = new _GreaterThan();
+  static final Condition lessThanOrEqual = new _LessThanOrEqual();
+  static final Condition always = new _Always();
+  static final Condition unpredictable = new _Unpredictable();
+
+  static final Map<int, Condition> _valueToCondition = <int, Condition>{
```
Comment (matanlurey) [inline arm7_tdmi/lib/src/instruction/arm/condition.dart]: Const once you do above.

File: arm7_tdmi/lib/src/instruction/arm/condition.dart [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+  String get name;
+
+  /// Returns true iff the state of [registers] passes this [Condition].
+  bool passes(ProgramStatusRegisters registers);
+
+  @override
+  String toString() => name;
+}
+
+class _Equal implements Condition {
```
Comment (matanlurey) [inline arm7_tdmi/lib/src/instruction/arm/condition.dart]: nit: final is actually a little more useful than a getter, here and below.

i.e.

```dart
@override
final name = 'EQ';
```

Comment (kharland) [inline arm7_tdmi/lib/src/instruction/arm/condition.dart]: Done

**Pull request merged**